lib/service/service.go

```diff
@@ -3654,18 +3654,18 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 				HandlerWithConnInfo: alpnproxy.ExtractMySQLEngineVersion(dbProxyServer.MySQLProxy().HandleConnection),
```

Comment:
> Maybe worth having a test for this case where there is an extra mysql engine version in the proto list.

Do you mean having both (MySQL + Ping suffix and MySQL version)? If yes, not sure if it is necessary since the PR doesn't change any of the router/mux.

> on second thought, have we considered similar approach for ping protocol (like using an extra protocol) instead of doubling each protocol? curious what pros and cons comparing current approach

[Yeah, It was discussed on the RFD if you want to take a look.](https://github.com/gravitational/teleport/pull/15152#discussion_r938097049)

Reply:
cool. That makes sense. So it will break if new client -> old server, cuz client can't tell if server multiplex ping or not